docs/configuration_files.rst

```diff
@@ -0,0 +1,350 @@
+          s: 250
+          size: 1000
+        hist, bin_edges: !numpy.histogram
+          a: $halos.circular_velocities
+          density: True
+
+  or a multi-column assignment
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        circular_velocities: !scipy.stats.maxwell.rvs
+          s: 250
+          size: 1000
+        histogram: !numpy.histogram
+          a: $halos.circular_velocities
+          density: True
+
+
+* `Referencing tables: table.init and table.complete dependencies`.
+
+  There are times when your function depends on tables. In these
+  cases, you need to ensure the referenced table has the necessary content and is not empty.
+
+  Example: you want to perform a very simple abundance matching, i.e. painting galaxies within your halos.
+  You can create two tables ``halos`` and ``galaxies`` storing the halo mass and galaxy luminosities.
+  Then you can stack these two tables and store it in a third table called ``matching``.
+
+  `Beware`: referencing tables is a bit different to variables or columns
+  -- where you simply tag a dollar sign.
+
+  The challenge: let ``tableA`` be a table with columns ``c1`` and ``c2``.
+  In configuration language, ``tableA`` is the name of the job.
+  *That means, when executing the configuration file, the first thing that happens is call ``tableA``, second,  call ``tableA.c1`` and third, call ``tableA.c2``.*
+  If you used the dollar sign to reference your ``tableA`` inside a function, this function might be called before the job ``tableA`` is complete, and the table will be empty.
+
+  `The solution`: to correctly reference tables, initialise your table with ``.init``, specify their dependences with the keyword ``.depends``
+  and ensure the tables are completed before calling the function with ``.complete``. Our example:
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        halo_mass: !numpy.random.uniform
+          low: 1.0e8
+          high: 1.0e14
+          size: 20
+      galaxies:
+        luminosity: !numpy.random.uniform
+          low: 0.05
+          high: 10.0
+          size: 20
+      matching:
+        .init: !astropy.table.vstack
+          tables: [ $halos, $galaxies ]
+          .depends: [ halos.complete, galaxies.complete ]
+
+
+Import objects
+^^^^^^^^^^^^^^
+* The SkyPy configuration syntax allows objects to be imported directly from external
+  (sub)modules using the ``!`` tag and providing neither a list of arguments or a
+  dictionary of keywords. For example, this enables the import and usage of any Astropy cosmology:
+
+  .. code:: yaml
+
+    cosmo1: !astropy.cosmology.Planck13 # import the Planck13 object and bind to cosmo1
+    cosmo2: !astropy.cosmology.default_cosmology.get [] # call default_cosmology.get() and bind to cosmo2
+
+
+
+Cosmology
+^^^^^^^^^
+
+* `Define parameters`: parameters are variables that can be modified at execution.
+
+  For example,
+
+  .. code:: yaml
+
+    parameters:
+      hubble_constant: 70
+      omega_matter: 0.3
```

Comment:
"Parameters" should probably be it's own section (it has nothing explicitly to do with "cosmology") with at least a brief explanation of what is different between a parameter and a normal variable